core/src/test/java/org/elasticsearch/env/NodeEnvironmentTests.java

```diff
@@ -300,7 +300,7 @@ public void run() {
     @Test
     public void testCustomDataPaths() throws Exception {
         String[] dataPaths = tmpPaths();
-        NodeEnvironment env = newNodeEnvironment(dataPaths, Settings.EMPTY);
+        NodeEnvironment env = newNodeEnvironment(dataPaths, "/tmp", Settings.EMPTY);
```

Comment:
This doesn't create any files here, it just checks and resolves paths (unit test only)